packages/studio-ui/src/components/IFramePortal.tsx

```diff
@@ -42,21 +44,62 @@ export default function IFramePortal(
   );
 }
 
-function useParentDocumentStyles(iframeDocument: Document | undefined) {
+function useInjectIframeCss(iframeDocument: Document | undefined) {
+  const [componentTree, getComponentMetadata, activePage, fileMetadatas] =
+    useStudioStore((store) => [
+      store.actions.getComponentTree(),
+      store.fileMetadatas.getComponentMetadata,
+      store.pages.activePageName,
+      store.fileMetadatas.UUIDToFileMetadata,
+    ]);
+
   useEffect(() => {
-    if (iframeDocument) {
-      const inlineStyles = document.head.getElementsByTagName("style");
-      const stylesheets = document.head.querySelectorAll(
-        'link[rel="stylesheet"]'
-      );
-      for (const el of [...inlineStyles, ...stylesheets]) {
-        iframeDocument.head.appendChild(el.cloneNode(true));
+    if (!iframeDocument) {
+      return;
+    }
+    componentTree?.forEach((component) => {
+      const metadataUUID = component.metadataUUID;
+      if (!metadataUUID) {
+        return;
       }
+      const cssImports = getComponentMetadata(metadataUUID).cssImports;
+      cssImports?.forEach((cssFilepath) => {
+        void dynamicImportFromBrowser(cssFilepath + "?inline").then(
```

Comment:
Even though we add the `inline` query param in the plugin, we need to have this `inline` query param here as well or else the function will not properly return the contents of the dynamic import (i.e. the import will happen, but it won't be accessible in the `then` since it seems unable to map back to the data if the id has changed).  I think it is just an idiosyncrasy of the dynamic import function.